### ISSUE TITLE
Trying SPFx Tutorial 5 without VS2015 installed

### ISSUE THREAD

**Issue by NPrice99** [SharePoint/sp-dev-docs#64]
Hi
I have set up my Windows 10 Machine for SPFx development.  I am using Visual Studio Code and not Visual Studio itself.

NB When you install Python (Setting up your machine) it needs .NET 2.0 to be alreafdy installed on your machine) 

When get to Step 3 (npm i --save jquery) I get a failure - MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it
 is installed elsewhere.  [C:\xxxxxxx\Documents\SPFx\Projects\jquery-webpart\node_modules\utf-8-validate\build\binding.sln]

After a bit of digging around  I found this - http://stackoverflow.com/questions/14278417/cannot-install-node-modules-that-require-compilation-on-windows-7-x64-vs2012

So I downloaded the Microsoft Build Tools 2013 (http://www.microsoft.com/en-us/download/details.aspx?id=40760) 

I set the global flag to use the 2013 version. (npm config set msvs_version 2013 --global)

The above error went away when I ran npm i --save jquery again but it was replaced with C:\xxxxxx\Documents\SPFx\Projects\jquery-webpart\node_modules\bufferutil\build\bufferutil.vcxproj(20,3): error MSB4019: The imported project "C:\Microsoft.Cpp.Default.props" was not found. Confirm that the
path in the <Import> declaration is correct, and that the file exists on disk.

Further Info - gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\12.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\xxxxxxxxxxx\AppData\Roaming\npm\node_modules\npm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)

Any Ideas how I get jquery to compile ??

Thanks

Nigel

**Comment by gavinbarron** [comment]
There's two easy steps to get things building without Visual Studio

0. `npm -g install npm@next`
0. `npm install --global --production windows-build-tools`

This will fix up your version of npm and then install and configure Python 2.7, the C++ build tools and set the node_gyp setting as required

**Comment by NPrice99** [comment]
Hi

After a bit more digging I found that Microsoft.Cpp.Defaults.props  is not installed in C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\v140 (I found this here - https://github.com/nodejs/node-gyp/issues/671).  SO it looks as if the C++ compiler is not installed.  The C:\Microsoft.Cpp.Default.props problem is because the file would be referenced as <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props"> and the there is no VCTargetsPath which is part of msbuild.

Regards

Nigel
